File: Python/LeetCode27.py
class Solution:
    def removeElement(self, nums, val):
        n = len(nums)
        total = n
        j = 0
        for i in range(n):
            if val == nums[i]:
                total -= 1
            else:
                nums[j] = nums[i]
                j += 1
        return total

File: leetCode27.c
#include<stdio.h>

int removeElement(int* nums, int numsSize, int val){

	int k=0,i;
	int n = numsSize;
	for (i=0;i<numsSize;i++)
		if(nums[i] == val)
			n--;
		else
			nums[k++] = nums[i];
	return n;
}

int main(void)
{
	int nums[] = {1,2,3,3,3,4,4,5};
	int numsSize = 8;
	int val = 3;
	
	int newLen = removeElement(nums, numsSize, val);
	int i;
	for (i=0;i<newLen;i++)
		printf("%d ", nums[i]);
}
